lib/solidus_dev_support/testing_support/factories.rb

```diff
@@ -30,18 +30,35 @@ module TestingSupport
     module Factories
       def self.load_for(*engines)
         paths = engines.flat_map do |engine|
-          factories_file_or_folder = engine.root.glob('lib/*/testing_support/factories{,.rb}')
-          if factories_file_or_folder.size == 2
-            folder, file = factories_file_or_folder.partition(&:directory?).map(&:first).map { |path| path.to_s.gsub(engine.root.to_s, '') }
+          # Check if the extension has a lib/*/factories.rb. If it does, we emit a
+          # deprecation warning an just use it.
+          obsolete_factories_file = engine.root.glob('lib/*/factories.rb').first
+          if obsolete_factories_file.present?
             ActiveSupport::Deprecation.warn <<-WARN.squish, caller(4)
               SolidusDevSupport::TestingSupport::Factories.load_for() is automatically loading
-              all factories present into #{folder}. You should now safely remove #{file} if it
-              is only used to load ./factories content.
+              all factories present in #{obsolete_factories_file.dirname.to_s.gsub(engine.root.to_s, '')}/testing_support/factories/.
+              Please move the content of #{obsolete_factories_file.to_s.gsub(engine.root.to_s, '')} to that directory.
             WARN
 
-            engine.root.glob('lib/*/testing_support/factories/**/*_factory.rb')
+            [obsolete_factories_file]
```

Comment:
This check seems to have some overlap with its `else` branch, the both look for `factories.rb`. 
The mere presence of the file doesn't mean that the file is obsolete or necessarily using `require` instead of just defining factories, for that reason I think it's better to remove this `if`.
Also the if in the `else` branch will never be `true` because whenever we have `factories.rb` we'll fall in the code above.

All of this unless I missed something, which is certainly possible 😅

Reply:
nvm, the first branch is not including `testing_support`